### PR TITLE
C# sdk V2

### PR DESCRIPTION
**Pull request by Florimond**
See the list of 42 commits.

### CODE REVIEW

File: Emitter/Emitter.Subscribe.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        public ushort On(string key, string channel, MessageHandler handler, int last)
+        {
+            // Register the handler
+            this.Trie.RegisterHandler(channel, handler);
+
+            // Subscribe
+            return this.Client.Subscribe(new string[] { FormatChannel(key, channel, Options.WithLast(last)) }, new byte[] { MqttMsgBase.QOS_LEVEL_AT_MOST_ONCE });
+        }
+
+        public ushort Subscribe(string channel, MessageHandler handler, params string[] options)
```
Comment (kelindar) [inline Emitter/Emitter.Subscribe.cs]: Add comments to functions.

File: Emitter/Emitter.Subscribe.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        /// <summary>
+        /// Asynchronously subscribes to a particular channel of emitter.io service.
+        /// </summary>
+        /// <param name="key">The key to use for this subscription request.</param>
+        /// <param name="channel">The channel to subscribe to.</param>
+        /// <param name="handler">The callback to be invoked every time the message is received.</param>
+        /// <param name="last">The last x messages to request when we subcribe.</param>
+        /// <returns>The message identifier for this operation.</returns>
+        [Obsolete("The On method is obsolete, please use the Subscribe method with the WithLast option instead.")]
+        public ushort On(string key, string channel, MessageHandler handler, int last)
```
Comment (kelindar) [inline Emitter/Emitter.Subscribe.cs]: Let's bump the major version and actually delete the code.

File: Emitter/Emitter.Subscribe.cs (hunk 1 above)
Comment (Florimond) [reply Emitter/Emitter.Subscribe.cs]: See new PR

File: Emitter/Emitter.Subscribe.cs (hunk 2 above)
Comment (Florimond) [reply Emitter/Emitter.Subscribe.cs]: See new PR

**Pull request closed without merging**